index.bs

```diff
@@ -12374,6 +12374,56 @@ described in the previous section).
             |rejectedPromise| to the callback function's return type.
 </div>
 
+Some callback functions are instead used as constructors. Such callback functions must not have
+a return type that is a [=promise type=].
+
+<div algorithm>
+
+    To <dfn id="construct-a-callback-function" export>construct</dfn> a
+    [=callback function type=] value |callable| with
+    a list of arguments |arg|<sub>0..|n|−1</sub>, each of which is either
+    an IDL value or the special value “missing” (representing a missing optional argument),
+    perform the following steps.
+    These steps will either return an IDL value or throw an exception.
+
+    1.  Let |completion| be an uninitialized variable.
```

Comment:
So this algorithm is pretty similar to <https://heycam.github.io/webidl/#invoke-a-callback-function>.  I wonder whether it's worth trying to factor parts of it out...  I guess the main similarity is the preparation (after the basic sanity-check, which is different) and the argument conversions, right?

Maybe that's not enough to worry about it.